src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs

```diff
@@ -41,5 +42,26 @@ public static string FrameworkDescription
                 return s_frameworkDescription;
             }
         }
+
+        /// <summary>
+        /// Returns an opaque string that identifies the platform on which an app is running.
+        /// </summary>
+        /// <remarks>
+        /// The property returns a string that identifies the operating system, typically including version,
+        /// and processor architecture of the currently executing process.
+        /// Since this string is opaque, it is not recommended to parse the string into its constituent parts.
+        /// </remarks>
+        public static string RuntimeIdentifier
+        {
+            get
+            {
+                if (s_runtimeIdentifier == null)
+                {
+                    s_runtimeIdentifier = (string?)AppContext.GetData("RUNTIME_IDENTIFIER") ?? "unknown";
```

Comment:
What happens if someone does `AppContext.SetData("RUNTIME_IDENTIFIER", somethingThatsNotString);`.  Then accessing RuntimeIdentifier would fail with a cast exception, right?  I think it'd be better to use an `as`, e.g.
What you have is fine.  It could also be:
```C#
public static string RuntimeIdentifier =>
    s_runtimeIdentifier ??= (AppContext.GetData("RUNTIME_IDENTIFIER") as string) ?? "unknown";
```

Reply:
We do a throwing cast in all other similar places that call `AppContext.GetData`

Reply:
Maybe we should fix those, too :)

Reply:
I fixed this call, and the FrameworkDescription above, since it is in the same file.

Reply:
That makes it inconsistent with the other place we check `FX_PRODUCT_VERSION` in Environment.cs ... it would be nice to have all `AppContext.GetData` string lookups done in a consistent way.

Reply:
Would you prefer to me to that in this PR or in a separate PR? (or change these two places back to a cast?)

Reply:
This PR is fine.